lib/scala/project-manager/src/main/scala/org/enso/projectmanager/service/validation/ModuleNameValidator.scala

```diff
@@ -0,0 +1,68 @@
+package org.enso.projectmanager.service.validation
+
+import cats.MonadError
+import cats.implicits._
+import org.enso.pkg.validation.{InvalidNameError, NameValidation}
+import org.enso.projectmanager.service.validation.ModuleNameValidator.ValidationFailure
+
+/** MTL implementation of the module name validator. */
+class ModuleNameValidator[F[_, _]](implicit
+  M: MonadError[F[ValidationFailure, *], ValidationFailure]
+) extends ProjectValidator[F, ValidationFailure] {
+
+  /** Validates a module name.
+    *
+    * @param name the module name
+    * @return either validation failure or success
+    */
+  override def validate(name: String): F[ValidationFailure, Unit] =
+    M.fromEither {
+      NameValidation
+        .validateName(name)
+        .leftMap(toValidationFailure)
+        .map(_ => ())
+    }
+
+  /** Convert project name error to validation error. */
+  private def toValidationFailure(error: InvalidNameError): ValidationFailure =
+    error match {
+      case InvalidNameError.Empty =>
+        ValidationFailure.EmptyName
+      case InvalidNameError.ShouldStartWithCapitalLetter =>
+        ValidationFailure.NameShouldStartWithCapitalLetter
+      case InvalidNameError.ContainsInvalidCharacters(invalidCharacters) =>
+        ValidationFailure.NameContainsForbiddenCharacter(invalidCharacters)
+      case InvalidNameError.ShouldBeUpperSnakeCase(validName) =>
+        ValidationFailure.NameShouldBeUpperSnakeCase(validName)
+    }
+
+}
+
+object ModuleNameValidator {
+
+  /** Base trait for validations failures. */
+  sealed trait ValidationFailure
```

Comment:
Why do we have two `ValidationFailure` traits? One here and another in `ProjectNameValidator`?